### PR TITLE
Add the beginner and customization tutorials for AutoMM

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Add the beginner tutorial for AutoMM.
2. Add the customization tutorial for AutoMM.
3. Simplify some Mixup related keys in configurations.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1861-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1861/1/index.html

File: docs/tutorials/text_prediction/automm_customization.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+The batch size for each GPU. 
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 8})
+# use batch size 16 per GPU
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 16})
+```
+
+### env.batch_size
+The batch size to use in each step of training. If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, we accumulate gradients to reach the effective `env.batch_size` before performing one optimization step.
```
Comment (bryanyzhu) [inline docs/tutorials/text_prediction/automm_customization.md:173]: If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, does it have to be multiples of `env.per_gpu_batch_size * env.num_gpus`. For example, if `per_gpu_batch_size=2`, `num_gpus=8`, when `batch_size` is 20, I'm curious what will gradient accumulation do?

File: docs/tutorials/text_prediction/automm_customization.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+Half precision, or mixed precision, is the combined use of 32 and 16 bit floating points to reduce memory footprint during model training. This can result in improved performance, achieving +3X speedups on modern GPUs.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.precision": 16})
+# use bfloat16
+predictor.fit(hyperparameters={"env.precision": "bf16"})
+```
+
+### env.num_workers
+The number of worker processes used by Pytorch dataloader in training.
```
Comment (bryanyzhu) [inline docs/tutorials/text_prediction/automm_customization.md:202]: Can we write some suggestions here, like depending on your number of CPU cores or number of GPUs, what number of worker process should the user adopt to maximize throughput?

File: docs/tutorials/text_prediction/automm_customization.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+See [here](https://pytorch-lightning.readthedocs.io/en/stable/accelerators/gpu.html#distributed-modes) for more details.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.strategy": "ddp_spawn"})
+# use ddp during training
+predictor.fit(hyperparameters={"env.strategy": "ddp"})
+```
+
+## Model
```
Comment (bryanyzhu) [inline docs/tutorials/text_prediction/automm_customization.md:235]: Are we leaving `model` and `data` for next PR or they have too much content to be put in a tutorial?

Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm_customization.md:236]: Just added.

File: docs/tutorials/text_prediction/automm_customization.md (hunk 2 above)
Comment (zhiqiangdon) [reply docs/tutorials/text_prediction/automm_customization.md:203]: Good suggestion.

**Comment by szha** [comment]
Job PR-1861-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1861/5/index.html

File: docs/tutorials/multimodal/customization.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
+# Customize AutoMM
+:label:`sec_automm_customization`
+
+AutoMM has a powerful yet easy-to-use configuration design.
+This tutorial walks you through various AutoMM configurations to empower you the customization flexibility. Specifically, AutoMM configurations consist of four parts: 
+- optimization
```
Comment (sxjscience) [inline docs/tutorials/multimodal/customization.md:6]: I guess we need to add an empty line above so the list can be appropriately rendered

**Comment by szha** [comment]
Job PR-1861-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1861/6/index.html

**Comment by szha** [comment]
Job PR-1861-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1861/7/index.html

**Comment by sxjscience** [comment]
We can consider to remove the automm tutorial in text to accelerate the build time. What do you think?

**Review by bryanyzhu — APPROVED** [review]

**Review by sxjscience — APPROVED** [review]
